src/libcore/lib.rs

```diff
@@ -117,6 +117,7 @@
 #![feature(track_caller)]
 #![feature(transparent_unions)]
 #![feature(unboxed_closures)]
+#![cfg_attr(not(bootstrap), feature(unsized_fn_params))]
 #![feature(unsized_locals)]
```

Comment:
`unsized_locals` should not be needed any more, so can you gate this by `cfg_attr(bootstrap, ...)`?

Reply:
If I do this change in `libcore` I get ...

```
Building stage1 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling cc v1.0.52
   Compiling core v0.0.0 (/home/santiago/src/oss/rust2/src/libcore)
   Compiling libc v0.2.69
   Compiling autocfg v0.1.7
   Compiling std v0.0.0 (/home/santiago/src/oss/rust2/src/libstd)
   Compiling hashbrown v0.6.2
   Compiling compiler_builtins v0.1.28
   Compiling backtrace-sys v0.1.37
   Compiling unwind v0.0.0 (/home/santiago/src/oss/rust2/src/libunwind)
error[E0161]: cannot move a value of type T: the size of T cannot be statically determined
   --> src/libcore/mem/mod.rs:162:33
    |
162 |     unsafe { intrinsics::forget(t) }
    |                                 ^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0161`.
error: could not compile `core`.

To learn more, run the command again with --verbose.
command did not execute successfully: "/home/santiago/src/oss/rust2/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "-Zbinary-dep-depinfo" "-j" "8" "--release" "--features" "panic-unwind backtrace compiler-builtins-c" "--manifest-path" "/home/santiago/src/oss/rust2/src/libtest/Cargo.toml" "--message-format" "json-render-diagnostics"
expected success, got: exit code: 101
failed to run: /home/santiago/src/oss/rust2/build/bootstrap/debug/bootstrap build --stage 1 src/libstd
Build completed unsuccessfully in 0:02:00
```

Reply:
Something is wrong then. The unsound feature must not be used in libcore/liballoc.

Reply:
If I'm not wrong this is the last missing bit to fix but I don't understand it exactly.
Wasn't libcore/liballoc already using that?. Isn't fixing some of this stuff for a different PR? or I may not be understanding what you're suggesting?.

Reply:
> Wasn't libcore/liballoc already using that?

Using what?

libstd/liballoc (and all of rustc really except for the test suite) should only be using the sound version of the feature. After all, we don't want unsound code in our standard library or the compiler.  So clearly `unsized_locals` should not be enabled. Does this make sense so far?

However, the bootstrap compiler does not understand `unsized_fn_params`, so we still have to enable the unsound `unsized_locals` *but only for the bootstrap compiler*.

Reply:
The error here looks like this is just an unsized parameter, not an arbitrary unsized local. So probably, your PR has a bug somewhere where it is still checking the wrong feature gate at some place.